tutorial/007_pruning.py

```diff
@@ -0,0 +1,94 @@
+# Set up the median stopping rule as the pruning condition.
+study = optuna.create_study(pruner=optuna.pruners.MedianPruner())
+study.optimize(objective, n_trials=20)
+
+###################################################################################################
+# Executing the script above:
+#
+# .. code-block:: console
+#
+#          $ python prune.py
+#          [I 2020-06-12 16:54:23,876] Trial 0 finished with value: 0.3157894736842105 and parameters: {'alpha': 0.00181467547181131}. Best is trial 0 with value: 0.3157894736842105.
+#          [I 2020-06-12 16:54:23,981] Trial 1 finished with value: 0.07894736842105265 and parameters: {'alpha': 0.015378744419287613}. Best is trial 1 with value: 0.07894736842105265.
+#          [I 2020-06-12 16:54:24,083] Trial 2 finished with value: 0.21052631578947367 and parameters: {'alpha': 0.04089428832878595}. Best is trial 1 with value: 0.07894736842105265.
+#          [I 2020-06-12 16:54:24,185] Trial 3 finished with value: 0.052631578947368474 and parameters: {'alpha': 0.004018735937374473}. Best is trial 3 with value: 0.052631578947368474.
+#          [I 2020-06-12 16:54:24,303] Trial 4 finished with value: 0.07894736842105265 and parameters: {'alpha': 2.805688697062864e-05}. Best is trial 3 with value: 0.052631578947368474.
+#          [I 2020-06-12 16:54:24,315] Trial 5 pruned.
+#          [I 2020-06-12 16:54:24,355] Trial 6 pruned.
+#          [I 2020-06-12 16:54:24,511] Trial 7 finished with value: 0.052631578947368474 and parameters: {'alpha': 2.243775785299103e-05}. Best is trial 3 with value: 0.052631578947368474.
+#          [I 2020-06-12 16:54:24,625] Trial 8 finished with value: 0.1842105263157895 and parameters: {'alpha': 0.007021209286214553}. Best is trial 3 with value: 0.052631578947368474.
+#          [I 2020-06-12 16:54:24,629] Trial 9 pruned.
+#          ...
+#          ``Trial 5 pruned.``, etc. in the log messages means several trials were stopped
+#          before they finished all of the iterations.
+#
+
+###################################################################################################
+# Integration Modules for Pruning
+# -------------------------------
+# To implement pruning mechanism in much simpler forms, Optuna provides integration modules for the following libraries.
+#
+# - XGBoost: :class:`optuna.integration.XGBoostPruningCallback`
+# - LightGBM: :class:`optuna.integration.LightGBMPruningCallback`
+# - Chainer: :class:`optuna.integration.ChainerPruningExtension`
+# - Keras: :class:`optuna.integration.KerasPruningCallback`
+# - TensorFlow :class:`optuna.integration.TensorFlowPruningHook`
+# - tf.keras :class:`optuna.integration.TFKerasPruningCallback`
+# - MXNet :class:`optuna.integration.MXNetPruningCallback`
+# - PyTorch Ignite :class:`optuna.integration.PyTorchIgnitePruningHandler`
+# - PyTorch Lightning :class:`optuna.integration.PyTorchLightningPruningCallback`
+# - FastAI :class:`optuna.integration.FastAIPruningCallback`
```

Comment:
This list is not up-to-date, but I think we can accept as it is. It will be replaced with a link to the `optuna.integration` reference as suggested in https://github.com/optuna/optuna/issues/1559.

Reply:
Resolved by https://github.com/optuna/optuna/pull/1543/commits/84a69064e330ed068d602089632a2235b7c2abd8.